app/src/userpages/components/DataUnionPage/Item.jsx

```diff
@@ -378,11 +388,73 @@ const Item = ({ product, stats }: Props) => {
         setIsOpen(!isOpen)
     }, [isOpen])
 
+    const validate = useCallback((product) => {
+        const invalidFields = validateProduct(product, isEthIdentityRequired)
+        const errors = Object.keys(invalidFields)
+            .filter((field) => !!invalidFields[field])
+            .map((field) => field)
+
+        if (errors.length > 0) {
+            errors.forEach((field) => {
+                Notification.push({
+                    title: validationErrors[field],
+                    icon: NotificationIcon.ERROR,
+                })
+            })
+
+            return false
+        }
+
+        if (isEthereumAddress(product.beneficiaryAddress)) {
+            const { active: activeMembers } = (stats && stats.memberCount) || {}
+
+            if ((activeMembers || 0) < dataUnionMemberLimit) {
+                Notification.push({
+                    title: `The minimum community size for a Data Union is ${dataUnionMemberLimit === 1 ?
+                        'one member' :
+                        `${numberToText(dataUnionMemberLimit)} members`
+                    }.`,
+                    icon: NotificationIcon.ERROR,
+                })
+                return false
+            }
+        }
+
+        return true
+    }, [isEthIdentityRequired, stats])
+
+    const publish = useCallback(async (productId) => wrapPublish(async () => {
+        const product = await getProductById(productId || '')
```

Comment:
Here we have to fetch the whole product because the product listing is not complete (and possibly out of date)